src/wallet-user/trust-protection/VaultRecoveryProcessTab.tsx

```diff
@@ -0,0 +1,267 @@
+            getBalances({ api, accountId: recoveredVaultAddress })
+                .then(balances => {
+                    setVaultBalances(balances.filter(balance => Number(balance.balance.toNumber()) > 0))
+                })
+        }
+    }, [ vaultBalances, setVaultBalances, recoveredVaultAddress, api ])
+
+    const { control, formState: { errors } } = useForm<FormValues>({
+        defaultValues: {
+            legalOfficer: null
+        }
+    });
+
+    const clearFormCallback = useCallback(() => {
+        setRecoveredCoinBalance(null);
+        setAsyncSignAndSubmit(null);
+        setSignAndSubmitError(false);
+        setLegalOfficer(null);
+        setStatus(Status.IDLE);
+    }, [ setRecoveredCoinBalance, setAsyncSignAndSubmit, setSignAndSubmitError ])
+
+    const amountToRecover = useMemo<PrefixedNumber>(() => recoveredCoinBalance?.balance ?
+            recoveredCoinBalance?.balance :
+            new PrefixedNumber("0", MILLI),
+        [ recoveredCoinBalance ])
+
+    const onTransferSuccessCallback = useCallback(async (_id: string, submittable: SuccessfulTransaction) => {
+        const axios = axiosFactory!(legalOfficer!);
+        const vaultApi = new VaultApi(axios, legalOfficer!);
+        const requesterAddress = accounts!.current!.address;
+        const blockHeader = await api!.rpc.chain.getHeader(submittable.block);
+        await vaultApi.createVaultTransferRequest({
+            amount: amountToRecover.convertTo(LGNT_SMALLEST_UNIT).coefficient.unnormalize(),
+            destination: targetVaultAddress!,
+            block: blockHeader.number.toString(),
+            index: submittable.index,
+            requesterAddress,
+        });
+        clearFormCallback();
+        refresh();
+    }, [ accounts, api, axiosFactory, legalOfficer, refresh, targetVaultAddress, clearFormCallback, amountToRecover ])
+
+    const recoverCoin = useCallback(async (amount: PrefixedNumber) => {
+        const signAndSubmit: AsyncSignAndSubmit = (setResult, setError) => {
+            return buildVaultTransferCall({
+                api: api!,
+                requesterAddress: recoveredAddress!,
+                destination: targetVaultAddress!,
+                recoveryConfig: recoveryConfig!,
+                amount: amount,
+            }).then(call => {
+                const unsubscriber = signAndSendAsRecovered({
+                    api: api!,
+                    signerId: accounts!.current!.address,
+                    callback: setResult,
+                    errorCallback: setError,
+                    recoveredAccountId: recoveredAddress!,
+                    call
+                });
+                return { unsubscriber }
+            }).catch(reason => {
+                setError(reason)
+                return {
+                    unsubscriber: Promise.resolve(() => {
+                    })
+                }
+            })
+        }
+        setAsyncSignAndSubmit(() => signAndSubmit)
```

Comment:
I think that this block can be rewritten into a more readable try-catch-await clause.

Reply:
Done.